integration/unplugin/src/index.ts

```diff
@@ -104,7 +144,12 @@ const civetUnplugin = createUnplugin((options: PluginOptions = {}) => {
           const sourceFile = program.getSourceFile(file)!;
           program.emit(
             sourceFile,
-            (filePath, content) => {
+            async (filePath, content) => {
+              const dir = path.dirname(filePath);
+              if (!pathExists(dir)) {
+                await fs.promises.mkdir(dir, { recursive: true });
+              }
```

Comment:
> Calling `fsPromises.mkdir()` when path is a directory that exists results in a rejection only when `recursive` is false.

So I don't think you need to check `pathExists`.

Reply:
Oh interesting, yep we can remove it then